src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -252,6 +253,17 @@ static ColumnWithTypeAndName readColumnWithDate32Data(std::shared_ptr<arrow::Chu
     DataTypePtr internal_type;
     bool check_date_range = false;
     /// Make result type Date32 when requested type is actually Date32 or when we use schema inference
+
+    DateTimeOverflowMode date_time_overflow_mode = DateTimeOverflowMode::THROW;
+
+    if (DB::CurrentThread::isInitialized())
```

Comment:
That's actually a neat trick that I didn't know before.

But to get the value of a setting, maybe it is cleaner to add it to the `FormatSettings` (which should be available to users of ArrowColumnToCHColumn).

@Avogar Do you have a preference/opinion here?

Reply:
But I'll still have to do the same thing here. Or am I wrong?

Reply:
> @Avogar Do you have a preference/opinion here?

Adding this setting to FormatSettings is ok.

> But I'll still have to do the same thing to retrieve the value here. Or am I wrong?

No, FormatSettings struct is passed and stored in input format classes like ParquetBlockInputFormat (and similar for Arrow and ORC), so you can add the value of this new setting as a constructir argument for ArrowColumnToCHColumn class. Liek it's done for `case_insensitive_matching` and `null_as_default` settings:
https://github.com/ClickHouse/ClickHouse/blob/966f1b6378e240652fc55c0f52ac6379a2654400/src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp#L499-L504